packages/compass-crud/src/components/readonly-document.tsx

```diff
@@ -48,13 +112,32 @@ class ReadonlyDocument extends React.Component<ReadonlyDocumentProps> {
     this.props.copyToClipboard?.(this.props.doc);
   };
 
+  /**
+   * Handle clicking the expand all button.
+   */
+  handleExpandAll = () => {
+    const { doc } = this.props;
+    // Update the doc directly - the components internal state will update via events
+    if (doc.expanded) {
+      doc.collapse();
+    } else {
+      doc.expand();
+    }
+  };
+
   /**
    * Get the elements for the document.
    *
    * @returns {Array} The elements.
    */
   renderElements() {
-    return <DocumentList.Document value={this.props.doc} />;
+    return (
+      <DocumentList.Document
+        value={this.props.doc}
+        // Provide extra whitespace for the expand button
+        extraGutterWidth={spacing[900]}
```

Comment:
If we want this to be very flexible in the component interface we probably should note that this value should match exactly the space left by hidden editing action buttons in the editable card, so that in case the editable card gets updated we can find this and update the value

Reply:
I don't think there a requirement that this offset match any other state as such. Just that it's enough for the expand button to not overlay the icons of the individual elements.